test/end-to-end/helpers/JSWait.js

```diff
@@ -42,7 +49,7 @@ class JSWait extends codecept_helper {
 
         } else {
             await helper.amOnPage(newUrl);
-            await helper.waitInUrl(newUrl);
+            await helper.waitInUrl(url);
```

Comment:
So this is to get around an issue that only relates to the IDAM page - where the URL that the test calls is basically just the root `/` with a language parameter and this results in `newUrl` looking like `/?lng=en`.

However, if the user isn't signed in yet, you land on the sign in page which has a URL something like `/sign-in?asdfasdf...` . So, if we await the `newUrl` here it fails. Waiting for the original `url` parameter isn't especially great since `/` will be valid on just literally any page, but then the sign-in step logic handles the relevant waiting for the sign-in page, and so it's safe to do and lets the test continue.

Reply:
And for non-IDAM URLs it works fine, checking the most important part of the URL before letting the step do any necessary language checks.